packages/components-native/src/InputFile/InputFile.tsx

```diff
@@ -0,0 +1,655 @@
+//  not image or video
+//  reference: https://www.iana.org/assignments/media-types/media-types.xhtml
+const restrictedToTypes = ["application/*", "audio/*", "text/*"];
+
+interface FileToUpload {
+  sourceFile: SourceFile;
+  fileUpload: FileUpload;
+}
+
+type MediaSource = FileSource.gallery | FileSource.video | FileSource.camera;
+
+interface MediaSourceSettings {
+  readonly requestPermission: () => Promise<PermissionsType>;
+  readonly permitted: (permissions: PermissionsType) => boolean;
+  readonly capture: (
+    responseHandler: (response: ImagePickerResponse) => void,
+  ) => void;
+  readonly messages: {
+    permissionDeniedTitle: typeof messages.permissionFailedCamera;
+    permissionDeniedMessage: typeof messages.permissionFailedDescription;
+  };
+}
+
+const mediaTypes: Record<AllowedTypes, MediaType> = {
+  images: "photo",
+  videos: "video",
+  mixed: "mixed",
+};
+
+function useInputFileHooks() {
+  const { onLogError, isOnline } = useAtlantisContext();
+  return { isOnline, onLogError };
+}
+
+const IS_ANDROID_33_OR_GREATER =
+  Platform.OS === "android" && Platform.Version >= 33;
+
+function getMenuOptions(formatMessage: IntlShape["formatMessage"]) {
+  const menuOptions = {
+    images: [
+      {
+        title: formatMessage(messages.takePhoto),
+        value: FileSource.camera,
+      },
+      {
+        title: formatMessage(messages.chooseFromLibrary),
+        value: FileSource.gallery,
+      },
+    ],
+    videos: [
+      ...(IS_ANDROID_33_OR_GREATER
+        ? []
+        : [
+            {
+              title: formatMessage(messages.takeVideo),
+              value: FileSource.video,
+            },
+          ]),
+      {
+        title: formatMessage(messages.chooseFromLibrary),
+        value: FileSource.gallery,
+      },
+    ],
+    mixed: [
+      {
+        title: formatMessage(messages.takePhoto),
+        value: FileSource.camera,
+      },
+      ...(IS_ANDROID_33_OR_GREATER
+        ? []
+        : [
+            {
+              title: formatMessage(messages.takeVideo),
+              value: FileSource.video,
+            },
+          ]),
+      {
+        title: formatMessage(messages.chooseFromLibrary),
+        value: FileSource.gallery,
+      },
+      {
+        title: formatMessage(messages.browseFiles),
+        value: FileSource.document,
+      },
+    ],
+  };
+
+  return menuOptions;
+}
+
+/* eslint max-statements: ["error", 15] */
+export function InputFile({
+  type = "singleSelect",
+  allowedTypes = "mixed",
+  accessibilityLabel,
+  accessibilityHint,
+  imageQuality = "original",
+  onInputError,
+  onFileSelected,
+  getUploadParams,
+  onUploadStart,
+  onUploadProgress,
+  onUploadComplete,
+  onUploadError,
+}: InputFileProps): JSX.Element {
+  const { formatMessage } = useIntl();
+  const { isOnline, onLogError } = useInputFileHooks();
+  if (Platform.OS === "ios") {
+    console.warn("InputFile is not supported on iOS");
```

Comment:
Will be removed when iOS is supported. I could throw an error but figured this would be sufficient